src/com/sap/piper/cm/ChangeManagement.groovy

```diff
@@ -172,29 +172,150 @@ public class ChangeManagement implements Serializable {
+        }
+
+        deployToolDependencies = deployToolDependencies.trim()
+
+        /*
+            In case the configuration has been adjusted so that no deployToolDependencies are provided
+            we assume an image is used which contains already all dependencies.
+            In this case we don't invoke npm install and we run the image with the standard user
+            already, since there is no need for being root. Hence we don't need to switch user also
+            in the script.
+         */
+        boolean noInstall = deployToolDependencies.isEmpty()
+
+        Iterable cmd = ['#!/bin/bash -e']
+
+        if (! noInstall) {
+            cmd << "npm install --global ${npmInstallOpts} ${deployToolDependencies}"
+            cmd << "su ${osDeployUser}"
+        } else {
+            script.echo "[INFO] no deploy dependencies provided. Skipping npm install call. Assuning docker image '${docker?.image}' contains already the dependencies for performing the deployment."
+        }
+
+        Iterable params = []
+
+        boolean useConfigFile = true, noConfig = false
+
+        if (!deployConfigFile) {
+            useConfigFile = false
+            noConfig = !script.fileExists('ui5-deploy.yaml')
+        } else {
+            if (script.fileExists(deployConfigFile)) {
+              // in this case we will use the config file
+              useConfigFile = true
+            } else {
+                if (deployConfigFile == 'ui5-deploy.yaml') {
+                    // in this case this is most likely provided by the piper default config and
+                    // it was not explicitly configured. Hence we assume not having a config file
+                    useConfigFile = false
+                    noConfig = true
+                } else {
+                    script.error("Configured deploy config file '${deployConfigFile}' does not exists.")
+                }
+            }
+        }
+
+        if (noConfig) {
+            params += ['--noConfig'] // no config file, but we will provide our parameters
+        }
+
+        if (useConfigFile) {
+             params += ['-c', "\"" + deployConfigFile + "\""]
+        }
+
+        //
+        // All the parameters below encapsulated in an if statement might also be provided in a config file.
+        // In case they are empty we don't add to the command line and we trust in the config file.
+        // In case they are finally missing the fiori deploy toolset will tell us.
+        //
+
+        if (transportRequestId) {
+            params += ['-t', transportRequestId]
+        }
+
+        if (endpoint) {
+            params += ['-u', endpoint]
+        }
+
+        if (abapPackage) {
+            params += ['-p', abapPackage]
+        }
+
+        if (applicationName) {
+            params += ['-n' , applicationName]
+        }
+
+        if (client) {
+            params += ['-l', client]
+        }
+
+        if (desc) {
```

Comment:
This will always be true because of line 188:
`def desc = description ?: 'Deployed with Piper based on SAP Fiori tools'`

Reply:
done